File: backend/src/main/java/com/campus/Campus_Connect/features/event/dto/response/EventHistoryResponse.java
package com.campus.Campus_Connect.features.event.dto.response;

import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EventHistoryResponse {

    private List<EventHistoryItemResponse> live;

    private List<EventHistoryItemResponse> upcoming;

    private List<EventHistoryItemResponse> past;

}

File: android-app/app/src/main/java/com/example/campusconnect/feature/events/data/remote/response/EventHistoryResponse.kt
package com.example.campusconnect.feature.events.data.remote.response

import com.google.gson.annotations.SerializedName

data class EventHistoryResponse(
    @SerializedName("live")
    val live: List<EventHistoryItemResponse>,

    @SerializedName("upcoming")
    val upcoming: List<EventHistoryItemResponse>,

    @SerializedName("past")
    val past: List<EventHistoryItemResponse>
)